app/models/solr_document.rb

```diff
@@ -150,9 +150,37 @@ def extents
   end
 
   def fetch_html_safe(field)
+    return fetch_summary_note(field) if field == "summary_storage_note_ssm"
     fetch(field, []).map(&:html_safe)
   end
 
+  def fetch_summary_note(field)
+    note = fetch(field, [])
+    matches_abid_format = note.first.match(/^(?<location>.*: Boxes )(?:[A-Z]-\d{6}; )+/)
```

Comment:
I think this will error if note is an empty array. Then `note.first` will be `nil` and `match` will be undefined. It would probably be good to add a test for this case so we can see it fail and then fix that bug.

I think it will show that we need to check for that nil before we run the match. What about a refactor like

```
note = fetch(field, [])
if note.present?
  abid_matcher = note.first.match(...)
  if abid_matcher
    boxes = note.first.scan(/[A-Z]-\d{6}/).sort
    note = ["#{abid_matcher[:location]}#{boxes_to_range(boxes)}"]
  end
end note.map(&:html_safe)
```

This also has the benefits of expanding the postfix conditionals, and only running the map once.